client/reader/stream/reader-tag-sidebar/index.jsx

```diff
@@ -114,13 +115,13 @@ const ReaderTagSidebar = ( {
 				<div className="reader-tag-sidebar-stats">
 					<div className="reader-tag-sidebar-stats__item">
 						<span className="reader-tag-sidebar-stats__count">
-							{ formatNumberCompact( tagStats?.data?.total_posts ) }
+							{ formatNumber( tagStats?.data?.total_posts, localeSlug ) }
```

Comment:
I see `formatNumber()` from `@automattic/components` and `formatNumberCompact()` from `/lib` are quite different and the latter is custom while the former uses `Intl.NumberFormat`. While I prefer using `Intl.NumberFormat()` generally, I'm concerned about any unexpected differences or regressions that this change might cause. Have you done a comparison between the two? I think some tests may be necessary to ensure we don't break something.

Reply:
I've added some tests - see below for my comments on the differences! Only other difference I could find is that with the exact number, there is one less significant number in the digit (ie. 1000 becomes 1K instead of 1.0K). 

I've checked the contexts in which this is used, and I personally don't think that's a big deal - especially since `formatNumberCompact()` doesn't account for all locales, whereas `Intl.NumberFormat` doesn't.

Reply:
Would be nice to list all the places where this is affected, since we might want to check them with design folks and ensure that the change is alright.